django_apscheduler/jobstores.py

```diff
@@ -208,6 +208,21 @@ def lookup_job(self, job_id: str) -> Union[None, AppSchedulerJob]:
             return None
 
     def get_due_jobs(self, now) -> List[AppSchedulerJob]:
+        # Close old and unusable connections. Specifically:
+        # - Persistent connections past CONN_MAX_AGE.
+        # - Connections that errored out (e.g. due to the DB going down).
+        # If closed, the query in _get_jobs() will try to reconnect.
+        #
+        # Normally, Django calls this when a request starts and finishes,
+        # however we are (probably) not running in request context but in
+        # a continuous loop. So without this, the same connection would be
+        # used forever, and any error would never be recovered from (e.g.
+        # when the DB comes back up).
+        #
+        # get_due_jobs() seems like the most appropriate place for this,
+        # as it's called at the beginning of every scheduler iteration.
+        db.close_old_connections()
```

Comment:
Yes, but database connections can be closed by the backend database engine literally anytime and while we are processing code **anywhere** in the project (see the issue log for examples of where this issue has cropped up elsewhere as well).

We need to find a catch-all solution if possible.